openedx/core/djangoapps/course_groups/models.py

```diff
@@ -90,7 +90,7 @@ def save(self, *args, **kwargs):
 
         max_retries = 5
         for num_attempts in range(max_retries):
-            saved_membership = None
+            saved_membership = self.inner_read()
```

Comment:
With this change here, do you also need to set `saved_memembership` below? It seems like we're duplicating work.